internal/checkcontroller/reconcile.go

```diff
@@ -0,0 +1,518 @@
+		return fmt.Errorf("failed to update nodeto unschedulable: %w", err)
+	}
+
+	logger.V(1).Info("Node drained and marked as unschedulable")
+	return nil
+}
+
+// CreateRebooterPodIfNeeded creates a rebooter pod for the node with the given name if it does not already exist.
+func (r *CheckControllerReconciler) CreateRebooterPodIfNeeded(ctx context.Context, workerName, nodeName, namespace string) error {
+	logger := log.FromContext(ctx).WithName("CreateRebooterPodIfNeeded").WithValues("workerName", workerName, "nodeName", nodeName)
+
+	logger.V(1).Info("Checking for existing rebooter pod")
+	existingPod, err := r.GetRebooterPod(ctx, nodeName, namespace)
+	if err == nil && existingPod != nil {
+		logger.Info("Rebooter pod exists", "podName", existingPod.Name)
+		err = r.Delete(ctx, existingPod)
+		if err != nil {
+			return fmt.Errorf("failed to delete existing rebooter pod: %w", err)
+		}
+		return nil
+	}
+
+	logger.V(1).Info("Creating rebooter pod")
+	if err := r.CreateRebooterPod(ctx, workerName, nodeName, namespace); err != nil {
+		return fmt.Errorf("failed to create rebooter pod for node %s: %w", nodeName, err)
+	}
+
+	logger.V(1).Info("Rebooter pod created successfully")
+	return nil
+}
+
+// GetRebooterPod gets the rebooter pod for the node with the given name in the given namespace.
+func (r *CheckControllerReconciler) GetRebooterPod(ctx context.Context, nodeName, namespace string) (*corev1.Pod, error) {
+	pod := &corev1.Pod{}
+	err := r.Get(ctx, client.ObjectKey{Name: fmt.Sprintf("node-rebooter-%s", nodeName), Namespace: namespace}, pod)
+	if err != nil {
+		return nil, fmt.Errorf("error getting rebooter pod: %w", err)
+	}
+	return pod, nil
+}
+
+// SetNodeConditionIfNotExists sets a custom condition on the node with the given name if it does not already exist.
+func (r *CheckControllerReconciler) SetNodeConditionIfNotExists(
+	ctx context.Context,
+	node *corev1.Node,
+	conditionType corev1.NodeConditionType,
+	status corev1.ConditionStatus,
+	message string,
+) error {
+	logger := log.FromContext(ctx).WithName("SetNodeConditionIfNotExists").WithValues("nodeName", node.Name)
+
+	// The field node.Status.Conditions belongs to the status of the Node resource.
+	// In Kubernetes, the status is considered a "system-owned" object and cannot be
+	// modified using a regular Update call.
+	// Instead, changes to the status must be made using the Status().Update method.
+	for i, cond := range node.Status.Conditions {
+		if cond.Type == conditionType {
+			if cond.Status == status {
+				logger.V(1).Info(fmt.Sprintf("Node already has condition %s set to %s", conditionType, status))
+				return nil
+			}
+			logger.V(1).Info("Updating existing condition to %s", status)
+			node.Status.Conditions[i] = corev1.NodeCondition{
+				Type:    conditionType,
+				Status:  status,
+				Message: message,
+				LastTransitionTime: metav1.Time{
+					Time: time.Now(),
+				},
+				LastHeartbeatTime: metav1.Time{
+					Time: time.Now(),
+				},
+			}
```

Comment:
lets maybe create new var for that, to avoid duplication in the next lines (266-276)
```
newCond := corev1.NodeCondition{
				Type:    conditionType,
				Status:  status,
				Message: message,
				LastTransitionTime: metav1.Time{
					Time: time.Now(),
				},
				LastHeartbeatTime: metav1.Time{
					Time: time.Now(),
				},
			}
```
and reuse it in line 250 and 266?